applications/PfemFluidDynamicsApplication/custom_processes/model_start_end_meshing_for_fluids_process.hpp

```diff
@@ -403,38 +411,63 @@ namespace Kratos
 
+	    if( i_mp->IsNot(CONTACT)  ){
 
-	    for(ModelPart::NodesContainerType::iterator i_node = i_mp->NodesBegin() ; i_node != i_mp->NodesEnd() ; i_node++)
-	      {
-		rComputingModelPart.Nodes().push_back(*(i_node.base()));
-		// rComputingModelPart.AddNode(*(i_node.base())); // very slow!
-	      }
+	      for(ModelPart::NodesContainerType::iterator i_node = i_mp->NodesBegin() ; i_node != i_mp->NodesEnd() ; i_node++)
+		{
+		  rComputingModelPart.Nodes().push_back(*(i_node.base()));
+		}
 
-	    for(ModelPart::ConditionsContainerType::iterator i_cond = i_mp->ConditionsBegin() ; i_cond != i_mp->ConditionsEnd() ; i_cond++)
-	      {
-		rComputingModelPart.AddCondition(*(i_cond.base()));
-	      }
+	      for(ModelPart::ConditionsContainerType::iterator i_cond = i_mp->ConditionsBegin() ; i_cond != i_mp->ConditionsEnd() ; i_cond++)
+		{
+		  rComputingModelPart.AddCondition(*(i_cond.base()));
+		}
 
 	    
-	    for(ModelPart::ElementsContainerType::iterator i_elem = i_mp->ElementsBegin() ; i_elem != i_mp->ElementsEnd() ; i_elem++)
-	      {
-		rComputingModelPart.AddElement(*(i_elem.base()));
-	      }
+	      for(ModelPart::ElementsContainerType::iterator i_elem = i_mp->ElementsBegin() ; i_elem != i_mp->ElementsEnd() ; i_elem++)
+		{
+		  rComputingModelPart.AddElement(*(i_elem.base()));
+		}
+	    }
+	  }
+
+	  if( (i_mp->Is(RIGID)) ){ 
+ 
+
+	    if( i_mp->IsNot(CONTACT)  ){
+
+	      for(ModelPart::NodesContainerType::iterator i_node = i_mp->NodesBegin() ; i_node != i_mp->NodesEnd() ; i_node++)
+		{
+		  if(i_node->IsNot(FLUID))
+		     rComputingModelPart.Nodes().push_back(*(i_node.base()));
+		}
+
+	      for(ModelPart::ConditionsContainerType::iterator i_cond = i_mp->ConditionsBegin() ; i_cond != i_mp->ConditionsEnd() ; i_cond++)
+		{
+		  rComputingModelPart.AddCondition(*(i_cond.base()));
+		}
+
+	    
+	      for(ModelPart::ElementsContainerType::iterator i_elem = i_mp->ElementsBegin() ; i_elem != i_mp->ElementsEnd() ; i_elem++)
+		{
+		  rComputingModelPart.AddElement(*(i_elem.base()));
+		}
+	    }
 	  }
 	}
 
       //Sort
-      rComputingModelPart.Nodes().Sort();
-      rComputingModelPart.Elements().Sort();
-      rComputingModelPart.Conditions().Sort();     
+      // rComputingModelPart.Nodes().Sort();
+      // rComputingModelPart.Elements().Sort();
+      // rComputingModelPart.Conditions().Sort();     
       
       //Unique
-      rComputingModelPart.Nodes().Unique();
-      rComputingModelPart.Elements().Unique();
-      rComputingModelPart.Conditions().Unique();
+      // rComputingModelPart.Nodes().Unique();
+      // rComputingModelPart.Elements().Unique();
+      // rComputingModelPart.Conditions().Unique();
       
       if( EchoLevel > 1 )
 	std::cout<<"    [ SUBMODEL PART ["<<rComputingModelPart.Name()<<"] [Elems="<<rComputingModelPart.NumberOfElements()<<"|Nodes="<<rComputingModelPart.NumberOfNodes()<<"|Conds="<<rComputingModelPart.NumberOfConditions()<<"] ] "<<std::endl;
```

Comment:
Did you test if the pfem fluid still works after these modifications? If yes, for me it is ok

Reply:
There are no tests in the application. I have tested some cases but I can not ensure all cases.